rules/rollup.go

```diff
@@ -30,61 +29,54 @@ import (
 	"github.com/m3db/m3metrics/filters"
 	"github.com/m3db/m3metrics/generated/proto/schema"
 	"github.com/m3db/m3metrics/policy"
+	"github.com/m3db/m3metrics/rules/models"
 
 	"github.com/pborman/uuid"
 )
 
 var (
-	emptyRollupTarget RollupTarget
+	emptyRollupTarget rollupTarget
 
 	errRollupRuleSnapshotIndexOutOfRange = errors.New("rollup rule snapshot index out of range")
 	errNilRollupTargetSchema             = errors.New("nil rollup target schema")
 	errNilRollupRuleSnapshotSchema       = errors.New("nil rollup rule snapshot schema")
 	errNilRollupRuleSchema               = errors.New("nil rollup rule schema")
 )
 
-// RollupTargetView is a human friendly representation of a rollup rule target at a given point in time.
-type RollupTargetView struct {
-	Name     string
-	Tags     []string
-	Policies []policy.Policy
-}
-
-func newRollupTargetView(target RollupTarget) RollupTargetView {
-	return RollupTargetView{
+func newRollupTargetView(target rollupTarget) models.RollupTargetView {
+	return models.RollupTargetView{
 		Name:     string(target.Name),
 		Tags:     stringArrayFromBytesArray(target.Tags),
 		Policies: target.Policies,
 	}
 }
 
-// RollupTarget creates a rollup target from a rollup target view.
-func (rtv RollupTargetView) RollupTarget() RollupTarget {
-	return RollupTarget{
-		Name:     []byte(rtv.Name),
-		Tags:     bytesArrayFromStringArray(rtv.Tags),
-		Policies: rtv.Policies,
+func rollupTargetViewsToTargets(views []models.RollupTargetView) []rollupTarget {
+	targets := make([]rollupTarget, len(views))
+	for i, t := range views {
+		targets[i] = rollupTargetFromView(t)
 	}
+	return targets
 }
 
-func rollupTargetViewsToTargets(views []RollupTargetView) []RollupTarget {
-	targets := make([]RollupTarget, len(views))
-	for i, t := range views {
-		targets[i] = t.RollupTarget()
+func rollupTargetFromView(rtv models.RollupTargetView) rollupTarget {
```

Comment:
Can we make this a ctor of `rollupTarget` that takes in a `RollupTargetView`?

Reply:
sure. I generally think of a constructor as a public function that returns some new public type, interface or otherwise, but I can make a private constructor looking function here.

Reply:
Took another look. Looks like the reason that I did not create a function here called `newRollupTarget`, which would be the expect constructor name, is because there is already a constructor function called that which takes in a `schema.RollupTarget` struct. I can change this one to be named `newRollupTargetFromView`.